File: Talks/2022-09-Paradigm-Conf/filter_odds.hs
filterOdds xs  = filter odd xs
filterOdds1    = filter odd
filterOdds2    = filter (\e -> (mod e 2) == 1)
filterOdds3 xs = [ x | x <- xs, odd x ]

main :: IO ()
main = do
    print $ filterOdds  [1..5]
    print $ filterOdds1 [1..5]
    print $ filterOdds2 [1..5]
    print $ filterOdds3 [1..5]

File: Talks/2022-09-Paradigm-Conf/filter_odds.py
# Godbolt: https://godbolt.org/z/G3GeG799E

def filter_odds(lst):
    res = []
    for e in lst:
        if e % 2:
            res.append(e)
    return res

def filter_odds1(lst):
    return list(filter(lambda e: e % 2, lst))

def filter_odds2(lst):
    return [e for e in lst if e % 2]

lst = [1,2,3,4,5]

print(filter_odds (lst))
print(filter_odds1(lst))
print(filter_odds2(lst))
